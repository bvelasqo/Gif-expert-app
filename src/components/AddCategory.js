import React, {useState} from "react";
import PropTypes from "prop-types";

const AddCategory = ({setCategories}) => {
	const [inputValue, setInputValue] = useState("");

	const handleInputChange = (event) => {
		setInputValue(event.target.value);
	};

	const handleSubmit = (event) => {
		event.preventDefault();
		if (inputValue.trim().length > 2) {
			setCategories((categories) => [inputValue, ...categories]);
			setInputValue("");
		} else console.log("The category must have at least 3 characters");
	};

	return (
		<form onSubmit={handleSubmit}>
			<div>
				<div className="container-input">
					<input
						type="text"
						value={inputValue}
						placeholder="Add a category"
						onChange={handleInputChange}
					/>
				</div>
				<hr style={{width: "67%", borderColor:"white"}}/>
			</div>
		</form>
	);
};

AddCategory.prototype = {
	setCategories: PropTypes.func.isRequired,
};

export {AddCategory};
