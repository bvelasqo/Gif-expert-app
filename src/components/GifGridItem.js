import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({id, title, url}) => {
	return (
		<div className="gif_item animate__animated animate__zoomIn">
			<img
				key={id}
				src={url}
				alt={title}
				className="gif animate__animated animate__zoomIn"
			/>
			<p className="title_image animate__animated animate__zoomIn animate__delay">{title}</p>
		</div>
	);
};

GifGridItem.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export {GifGridItem};