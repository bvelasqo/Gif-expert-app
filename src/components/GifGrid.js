import React from "react";
import PropTypes from "prop-types";
import {useFetchGifs} from "../hooks/useFetchGifs";
import {GifGridItem} from "./GifGridItem";

const GifGrid = ({category}) => {
	const {data: gifs, loading} = useFetchGifs(category);

	return (
		<div className="main_gif_grid">
			<h3 className="category animate__animated animate__bounceIn">Category: {category}</h3>
			{/* loading con && */}
			{loading && <p>Loading...</p>}
			{/* loading con || */}
			{loading ? (
				<div className="animate__animated animate__flash">
					<img
						src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
						alt="loading"
					/>
				</div>
			) : (
				<div className="gif_grid">
					{gifs.map((image) => (
						<>
							<GifGridItem key={image.id} {...image} />
						</>
					))}
				</div>
			)}
		</div>
	);
};

GifGrid.prototype = {
	category: PropTypes.string.isRequired,
};

export default GifGrid;
