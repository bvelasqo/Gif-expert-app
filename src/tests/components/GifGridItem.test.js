import {GifGridItem} from "../../components/GifGridItem";
import Shallow from "enzyme/Shallow";

describe("GifGridItem", () => {
	const title = "Title";
	const url = "https://media1.giphy.com/media/l0HlW7QXQX5qQQq0U/200w.gif";
	const id = "l0HlW7QXQX5qQQq0U";

	let wrapper = Shallow(<GifGridItem id={id} title={title} url={url} />);

  // beforeEach is a hook that runs before each test is run (beforeEach is a function)
	beforeEach(() => {
    // Shallow render component before each test case to avoid re-rendering component each time test case is run (performance)
		wrapper = Shallow(<GifGridItem id={id} title={title} url={url} />);
	});

	test("should render correctly", () => {
    // toMatchSnapshot() is a method from jest-dom that allows us to test the component in a snapshot test format (see https://jestjs.io/docs/en/expect)
		expect(wrapper).toMatchSnapshot();
	});

	test("Should to have a <p> tag with the title", () => {
    // find() is a method from enzyme that allows us to find a specific element in the component
		const p = wrapper.find("p");
		expect(p.text().trim()).toBe(title);
	});

	test("Should to have a <img> tag with the url and alt", () => {
		const img = wrapper.find("img");
    // prop is a method from enzyme that allows us to test the component in a snapshot test format (see https://jestjs.io/docs/en/expect)
		expect(img.prop("src")).toBe(url);
    // toBe() is a method from jest-dom that allows us to test the component in a snapshot test format (see https://jestjs.io/docs/en/expect)
		expect(img.prop("alt").trim()).toBe(title);
	});

  test("Should to have animate__animated animate__zoomIn class", () => {
    // find() is a method from enzyme that allows us to find a specific element in the component
    const img = wrapper.find("img");
    //hasClass() is a method from enzyme that allows us to test the component in a snapshot test format (see https://jestjs.io/docs/en/expect)
    expect(img.hasClass("animate__animated animate__zoomIn")).toBe(true);
  });

  test("Shoul be a <div> tag", () => {
    // find() is a method from enzyme that allows us to find a specific element in the component
    const div = wrapper.find("div");
    // toBe() is a method from jest-dom that allows us to test the component in a snapshot test format (see https://jestjs.io/docs/en/expect)
    expect(div.length).toBe(1);
  });
});
