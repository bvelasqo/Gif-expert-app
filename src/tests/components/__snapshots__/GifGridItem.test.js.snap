// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should render correctly 1`] = `
<div
  className="gif_item animate__animated animate__zoomIn"
>
  <img
    alt="Title"
    className="gif animate__animated animate__zoomIn"
    key="l0HlW7QXQX5qQQq0U"
    src="https://media1.giphy.com/media/l0HlW7QXQX5qQQq0U/200w.gif"
  />
  <p
    className="title_image animate__animated animate__zoomIn animate__delay"
  >
    Title
  </p>
</div>
`;
